src/shared/validators.ts

```diff
@@ -0,0 +1,207 @@
+import { UInt4 } from 'bitwise/types';
+import { MAX_ADDR, encReqCodes, EMPTY_WALLET_UID, ASCII_REGEX } from '../constants';
+import { isUInt4, checksum } from '../util';
+
+export const validateValueExists = (arg: { [key: string]: any }) => {
+  const [key, [, value]] = Object.entries(arg);
+  if (!value) {
+    throw new Error(`${key} must be provided`);
+  }
+};
+
+export const validateIsUInt4 = (n?: number) => {
+  if (typeof n !== 'number' || !isUInt4(n)) {
+    throw new Error('Must be an integer between 0 and 15 inclusive');
+  }
+  return n as UInt4;
+};
+
+export const validateNAddresses = (n: number) => {
+  validateIsUInt4(n);
```

Comment:
This line doesn't seem necessary and would be bad if we ever decided to return more than 16 addresses in a future fw version

Reply:
I think we had that in the previous version? But yeah, I agree.

Reply:
991ebaa